api/v1alpha1/bundle_types.go

```diff
@@ -110,6 +113,10 @@ type HTTPSource struct {
 	Auth Authorization `json:"auth,omitempty"`
 }
 
+type OCIArtifactSource struct {
+	// Ref contains the reference to a OCI Artifact Image containing Bundle contents.
+	Ref string `json:"ref"`
```

Comment:
Since we're pulling the content directly, I assume a fully fleshed out version of this api would have to add fields for things like:

1) pull secret (presumbly a ref to a secret in the same NS)

and OLM itself would need a configuration for (or again respect the cluster's configuration for):
1) allowed/denied registries
2) image signatures
3) additional trusted CAs
4) proxy

some of these are applicable to the http and git source types also.